cob_twist_controller/include/cob_twist_controller/utils/moving_average.h

```diff
@@ -35,12 +35,12 @@ class MovingAverage
 {
     private:
         std::deque<double> s_;
-        std::deque<double> weighting_, temp_;
+        std::deque<double> weighting_;
 
         unsigned int size_;
 
     public:
-        MovingAverage(int size=3)
+        MovingAverage(unsigned int size=3)
```

Comment:
I've made MovingAverage class generic with templates. Additionally considered the things you've implemented there.